src/math/spherical_harmonics.cc

```diff
@@ -55,17 +56,12 @@ namespace rascal {
       double sin_theta = sqrt(1.0 - pow(cos_theta, 2));
       // Matrix_t assoc_legendre_polynom(max_angular + 1, max_angular +
       // 1);
-      // TODO(alex) make these into compact lm storage, and pre-allocate so
-      // it does not allocate a matrix every time it calls spherical harmonics
       Matrix_t assoc_legendre_polynom =
-          Matrix_t::Zero(max_angular + 1, max_angular + 1);
+          Matrix_t::Zero(max_angular + 1, max_angular + 2);
```

Comment:
Right, to simplify the derivative code I decided that we'd store associated Legendre polynomials for $m = l+1$, which mathematical convention sets equal to zero. (This works out very nicely, because it makes the raising operator applied to $Y^l_l$ give zero -- which gives you the constraint that the magnetic quantum number $m$ can only vary between $-l$ and $l$). Long story short, we store an extra row of zeros so we don't get nasty memory errors when trying to access $P^l_{l+1}$; we just get zero.